### PR TITLE
Use "real" GitUICommands instance

### LINKED ISSUE #11906: [NBug] No service for type 'ResourceManager.IHotkeySettingsLoader'...
### Environment

- Git Extensions 5.0.0.17897
- Build 2a3b78b868e7757d986c49bbfc03406b38f1ffe9
- Git 2.45.2.windows.1
- Microsoft Windows NT 10.0.22631.0
- .NET 8.0.8
- DPI 96dpi (no scaling)
- Portable: False

### Issue description

```
System.InvalidOperationException: No service for type 'ResourceManager.IHotkeySettingsLoader' has been registered.
   at GitUIPluginInterfaces.ServiceProviderExtensions.GetRequiredService[T](IServiceProvider provider) in C:\projects\gitextensions\src\plugins\GitUIPluginInterfaces\ServiceProviderExtensions.cs:line 50
   at GitUI.GitModuleForm.set_UICommands(IGitUICommands value) in C:\projects\gitextensions\src\app\GitUI\GitModuleForm.cs:line 61
   at GitUI.GitModuleForm..ctor(IGitUICommands commands, Boolean enablePositionRestore) in C:\projects\gitextensions\src\app\GitUI\GitModuleForm.cs:line 94
   at GitUI.GitExtensionsDialog..ctor(IGitUICommands commands, Boolean enablePositionRestore) in C:\projects\gitextensions\src\app\GitUI\GitExtensionsDialog.cs:line 20
   at GitUI.HelperDialogs.FormStatus..ctor(IGitUICommands commands, ConsoleOutputControl consoleOutput, Boolean useDialogSettings) in C:\projects\gitextensions\src\app\GitUI\HelperDialogs\FormStatus.cs:line 18
   at GitUI.HelperDialogs.FormProcess..ctor(IGitUICommands commands, ConsoleOutputControl outputControl, ArgumentString arguments, String workingDirectory, String input, Boolean useDialogSettings, String process) in C:\projects\gitextensions\src\app\GitUI\HelperDialogs\FormProcess.cs:line 24
   at GitUI.HelperDialogs.FormProcess..ctor(IGitUICommands commands, ArgumentString arguments, String workingDirectory, String input, Boolean useDialogSettings, String process) in C:\projects\gitextensions\src\app\GitUI\HelperDialogs\FormProcess.cs:line 56
   at GitUI.HelperDialogs.FormRemoteProcess..ctor(IGitUICommands commands, ArgumentString arguments) in C:\projects\gitextensions\src\app\GitUI\HelperDialogs\FormRemoteProcess.cs:line 33
   at GitUI.CommandsDialogs.RepoHosting.ForkAndCloneForm.Clone(IHostedRepository repo) in C:\projects\gitextensions\src\app\GitUI\CommandsDialogs\RepoHosting\ForkAndCloneForm.cs:line 377
   at GitUI.CommandsDialogs.RepoHosting.ForkAndCloneForm._cloneBtn_Click(Object sender, EventArgs e) in C:\projects\gitextensions\src\app\GitUI\CommandsDialogs\RepoHosting\ForkAndCloneForm.cs:line 308
   at System.Windows.Forms.Button.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnMouseUp(MouseEventArgs mevent)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ButtonBase.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(HWND hWnd, MessageId msg, WPARAM wparam, LPARAM lparam)
```

Clone new git repository from Github

### Steps to reproduce

Clone repository from GitHub

### Did this work in previous version of GitExtensions?

_No response_

### Diagnostics

_No response_

### PR DESCRIPTION
**Pull request by RussKie**
Resolves #11906

## Test methodology 

- manually

## Test environment(s) 

- GIT 
- Windows 

## Merge strategy

I agree that the maintainer squash merge this PR (if the commit message is clear).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Review by pmiossec — APPROVED** [review]
OK but I prefer a better naming because I'm a bit lost in namings

File: src/app/GitUI/CommandsDialogs/RepoHosting/ForkAndCloneForm.cs [hunk 1]
```diff
@@ -34,12 +34,14 @@ public partial class ForkAndCloneForm : GitExtensionsForm
...
         private const string UpstreamRemoteName = "upstream";
+        private readonly IGitUICommands _commands;
         private readonly IRepositoryHostPlugin _gitHoster;
         private readonly EventHandler<GitModuleEventArgs>? _gitModuleChanged;
 
-        public ForkAndCloneForm(IRepositoryHostPlugin gitHoster, EventHandler<GitModuleEventArgs>? gitModuleChanged)
+        public ForkAndCloneForm(IGitUICommands commands, IRepositoryHostPlugin gitHoster, EventHandler<GitModuleEventArgs>? gitModuleChanged)
         {
             _gitModuleChanged = gitModuleChanged;
+            _commands = commands;
```
[the remark this answers is not in the document]
Comment (RussKie) [reply src/app/GitUI/CommandsDialogs/RepoHosting/ForkAndCloneForm.cs:44]: I followed the existing naming pattern:
image

I get your point (it's valid), but we'll need to rename across the codebase, which isn't something I want to do here.

File: src/app/GitUI/CommandsDialogs/RepoHosting/ForkAndCloneForm.cs [hunk 2]
```diff
@@ -34,12 +34,14 @@ public partial class ForkAndCloneForm : GitExtensionsForm
         #endregion
 
         private const string UpstreamRemoteName = "upstream";
+        private readonly IGitUICommands _commands;
         private readonly IRepositoryHostPlugin _gitHoster;
         private readonly EventHandler<GitModuleEventArgs>? _gitModuleChanged;
 
-        public ForkAndCloneForm(IRepositoryHostPlugin gitHoster, EventHandler<GitModuleEventArgs>? gitModuleChanged)
+        public ForkAndCloneForm(IGitUICommands commands, IRepositoryHostPlugin gitHoster, EventHandler<GitModuleEventArgs>? gitModuleChanged)
```
Comment (mstv) [inline src/app/GitUI/CommandsDialogs/RepoHosting/ForkAndCloneForm.cs:41]:
```suggestion
        public ForkAndCloneForm(IServiceProvider serviceProvider, IRepositoryHostPlugin gitHoster, EventHandler<GitModuleEventArgs>? gitModuleChanged)
```

File: src/app/GitUI/CommandsDialogs/RepoHosting/ForkAndCloneForm.cs [hunk 3]
```diff
@@ -370,11 +372,9 @@ private void Clone(IHostedRepository repo)
                 return;
             }
 
-            GitUICommands uiCommands = new(GitUICommands.EmptyServiceProvider, new GitModule(null));
+            ArgumentString cmd = Commands.Clone(repo.CloneUrl, targetDir, _commands.Module.GetPathForGitExecution, depth: GetDepth());
```
Comment (mstv) [inline src/app/GitUI/CommandsDialogs/RepoHosting/ForkAndCloneForm.cs:375]:
```suggestion
            GitUICommands uiCommands = new(_serviceProvider, new GitModule(null));
```

File: src/app/GitUI/GitUICommands.cs [hunk 4]
```diff
@@ -1302,7 +1302,7 @@ public void StartCloneForkFromHoster(IWin32Window? owner, IRepositoryHostPlugin
         {
             WrapRepoHostingCall(TranslatedStrings.ForkCloneRepo, gitHoster, gh =>
             {
-                using ForkAndCloneForm frm = new(gh, gitModuleChanged);
+                using ForkAndCloneForm frm = new(this, gh, gitModuleChanged);
```
Comment (mstv) [inline src/app/GitUI/GitUICommands.cs:1305]:
```suggestion
                using ForkAndCloneForm frm = new(serviceProvider: this, gh, gitModuleChanged);
```

**Review by mstv — COMMENTED** [review]
Good catch!
(I wonder why most of us could not reproduce the exception.)

**Comment by pmiossec** [comment]
> (I wonder why most of us could not reproduce the exception.)

I was about (I worked on it and diagnosed the issue but I had to stop working on it and so was slower than @RussKie to propose the fix)

File: src/app/GitUI/CommandsDialogs/RepoHosting/ForkAndCloneForm.cs (hunk 2 above)
Comment (RussKie) [reply src/app/GitUI/CommandsDialogs/RepoHosting/ForkAndCloneForm.cs:41]: We don't do this for the forms, only for controls. I haven't given this too much thought, but on the surface it's probably a good change to make across the codebase.

**Pull request merged**